samples/GalleryApp/Models/SkillView.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+
+namespace GalleryApp
+{
+    /// <summary>
+    /// This class represents the a skill and its related xaml page information
+    /// </summary>
+    public class SkillView
+    {
+        private const string m_namespace = "GalleryApp";
+
+        /// <value> Get the page type string from JSON file </value>
+        public string PageTypeStr
+        {
+            get
+            {
+                return PageType.ToString();
+            }
+            set
+            {
+                PageType = System.Type.GetType(m_namespace + "." + value);
+                InitializeWithSkillDescriptor();
+            }
+        }
+
+        public string Name { get; private set; }
+        public string Description { get; private set; }
+
+        /// <value> Get the PageType for frame navigation </value>
+        public Type PageType { get; private set; }
+
+        /// <summary>
+        /// Get skill information from the corresponding skill descriptor object and assign to the related fields
+        /// </summary>
+        private void InitializeWithSkillDescriptor()
+        {
+            var page = Activator.CreateInstance(PageType) as ISkillViewPage;
+            var descriptor = page.GetSkillDescriptor();
+            Name = descriptor.Name;
+            Description = descriptor.Description;
+        }
+
+        /// <summary>
+        /// Constructor
```

Comment:
>Constructor [](start = 12, length = 11)

Is it needed?

Reply:
No. 

---
In reply to: [302649722](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/45#discussion_r302649722) [](ancestors = 302649722)